### PR TITLE
Fix tests when HOSTNAME is set, but empty string

### PR DESCRIPTION
**Pull request by hroncok**
See https://github.com/tox-dev/tox/pull/1616#issuecomment-658437312

This is a fixup of cfe66fd0fe8a81bd65f79f589486b7721632a4cb.

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: tests/unit/test_result.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def create_fake_pkg(tmpdir):
 @pytest.fixture()
 def no_hostname_envvar():
     hostname = os.getenv("HOSTNAME")
```
Comment (asottile) [inline tests/unit/test_result.py:22]: perhaps a better alternative is to use pytest's builtin `monkeypatch.setenv` ?

Comment (hroncok) [reply tests/unit/test_result.py:22]: Right, I forgot it exists. I can get to it tomorrow, I'm already away from keyboard.

Comment (hroncok) [reply tests/unit/test_result.py]: I've rebased the PR to use it, thanks for the suggestion.

**Comment by hroncok** [comment]
Thanks for the suggestion and merge!

**Comment by gaborbernat** [comment]
This now is done under 3.17.1, thanks for the contribution!

**Pull request merged**